pkg/registry/core/service/ipallocator/controller/repair.go

```diff
@@ -136,6 +145,7 @@ func (c *Repair) runOnce() error {
 		ip := net.ParseIP(svc.Spec.ClusterIP)
 		if ip == nil {
 			// cluster IP is corrupt
+			c.recorder.Eventf(&svc, v1.EventTypeWarning, "ClusterIPNotValid", "Cluster IP %s is not a valid IP; please recreate service", svc.Spec.ClusterIP)
```

Comment:
@gmarek - in the context of new event API, what is now the policy of adding new events?
I think those are added in this PR are generally useful, but I'm not sure if we should add them in the old API or maybe wait for the new one?

Reply:
We add them with the old API, as new one doesn't have proper client library implemented yet.